heft-plugins/heft-sass-plugin/src/SassPlugin.ts

```diff
@@ -43,84 +45,125 @@ export default class SassPlugin implements IHeftPlugin {
     });
 
     taskSession.hooks.run.tapPromise(PLUGIN_NAME, async (runOptions: IHeftTaskRunHookOptions) => {
-      // TODO: Handle watch mode
-      await this._runSassTypingsGeneratorAsync(taskSession, heftConfiguration, false);
+      await this._runSassTypingsGeneratorAsync(taskSession, heftConfiguration);
     });
+
+    taskSession.hooks.runIncremental.tapPromise(
+      PLUGIN_NAME,
+      async (runIncrementalOptions: IHeftTaskRunIncrementalHookOptions) => {
+        await this._runSassTypingsGeneratorAsync(taskSession, heftConfiguration, runIncrementalOptions);
+      }
+    );
   }
 
   private async _runSassTypingsGeneratorAsync(
     taskSession: IHeftTaskSession,
     heftConfiguration: HeftConfiguration,
-    isWatchMode: boolean
+    runIncrementalOptions?: IHeftTaskRunIncrementalHookOptions
   ): Promise<void> {
-    const sassConfiguration: ISassConfiguration = await this._loadSassConfigurationAsync(
+    taskSession.logger.terminal.writeVerboseLine('Starting sass typings generation...');
+    const sassProcessor: SassProcessor = await this._loadSassProcessorAsync(
       heftConfiguration,
       taskSession.logger
     );
-    const sassTypingsGenerator: SassProcessor = new SassProcessor({ sassConfiguration });
-    await sassTypingsGenerator.generateTypingsAsync();
-    if (isWatchMode) {
-      Async.runWatcherWithErrorHandling(
-        async () => await sassTypingsGenerator.runWatcherAsync(),
-        taskSession.logger
+    // If we have the incremental options, use them to determine which files to process.
+    // Otherwise, process all files. The typings generator also provides the file paths
+    // as relative paths from the sourceFolderPath.
+    let changedFilePaths: string[] | undefined;
+    if (runIncrementalOptions) {
+      changedFilePaths = [];
+      const filePaths: string[] = runIncrementalOptions.globChangedFiles(sassProcessor.inputFileGlob, {
+        cwd: sassProcessor.sourceFolderPath,
+        ignore: Array.from(sassProcessor.ignoredFileGlobs)
+      });
+      const deleteFilePromises: Promise<void>[] = [];
+      for (const filePath of filePaths) {
+        // Filter out and delete any files that are removed and all their output files
+        const absoluteFilePath: string = path.join(sassProcessor.sourceFolderPath, filePath);
+        if (runIncrementalOptions.changedFiles.get(absoluteFilePath)!.version === undefined) {
+          deleteFilePromises.push(
```

Comment:
Even if the `getOutputFilePaths()` call _shouldn't_ return thousands of entries and cause a stack overflow, it's better not to expand a dynamically sized array onto the stack.